services/src/main/java/org/keycloak/theme/DefaultThemeManager.java

```diff
@@ -276,6 +277,32 @@ private Map<Locale, Properties> getMessagesByLocale(String baseBundlename, Local
             }
         }
 
+        protected void addlocaleTranslations(Locale locale, Properties m) throws IOException {
+            for (String l : getProperties().getProperty("locales", "").split(",")) {
+                l = l.trim();
+                String key = "locale_" + l;
+                String label = m.getProperty(key);
+                if (label == null) {
+                    String rl = l;
+                    if (l.equals("zh-CN")) {
+                        rl = "zh-HANS";
+                    } else if (l.equals("zh-TW")) {
+                        rl = "zh-HANT";
+                    }
+                    Locale loc = Locale.forLanguageTag(rl);
+                    label = capitalize(loc.getDisplayName(locale), locale);
+                    if (!Objects.equals(loc, locale)) {
+                        label += " (" + capitalize(loc.getDisplayName(loc), loc) + ")";
```

Comment:
This kind of source order also seems something that could be very locale dependent. E.g. in Arabic you might want to reverse the order here. Perhaps we should add a label for the format of this? E.g. `locale_choice={0} ({1})`.

Reply:
Even in RTL languages, isn't the string just printed RTL? For now, I'd like to keep it as is, as it is an improvement to the current way IMHO

Reply:
Ok, let's file this as a possible enhancement as I don't want to hold up what is already a good improvement, let's log a follow up issue for this.

@Eng-Fouad @sharifrezvani since you maintain some of the RTL languages, I'd like to know what you think about this.

Reply:
Here is how it would look when printing locales in Arabic:

using LTR UI (notice how RTL locale is printed):

![ltr](https://github.com/keycloak/keycloak/assets/1194488/e3804343-22d8-4732-b29b-a0bf866e446d)

using RTL UI:

![rtl](https://github.com/keycloak/keycloak/assets/1194488/319075ad-d12c-4066-887c-28063b39f5e4)

Reply:
Sounds good, thanks for the feedback @Eng-Fouad